### DISCUSSION TITLE
ROCM 5.5 released

### DISCUSSION THREAD

**Discussion by iDeNoh** [vladmandic/sdnext#741] [General]
Looks like RDNA3 cards are fully supported AND it now supports memory attention with SDP, any chance anyone *slightly* more knowledgeable than me could build torch 2.0 with ROCM 5.5 from source or point me in the direction of reliable instructions to do so on my own? Really excited to give it a spin.

**Comment by vladmandic** [comment]
btw, you might also monitor torch 2.1 nighly builds and see if it pops up sometimes soon: <https://download.pytorch.org/whl/nightly/torch/>

**Reply by iDeNoh** [reply]
Lol, yea my eyes have been glued to that page for the last day 😔

**Reply by vladmandic** [reply]
well, torch team is not the fastest, i'd give it 2-3 weeks at least. 
so if you want to test earlier, yeah, building torch is the way to go.

**Reply by Soulreaver90** [reply]
Tried to compile my own but I dont know what I was doing and the compiled torch wouldn't work, plus it took like an hour so I don't want to do that again, lol.

**Reply by DistantThunder** [reply]
@vladmandic Thank you for your work!

I think it'd do a lot for your users if you could spare them the local compile woes and propose something at:

* https://hub.docker.com/r/rocm/pytorch 

by yourselves.

**Reply by vladmandic** [reply]
i'd love to, but AMD ROCm support here is community driven - I don't have test system with AMD GPU.

**Reply by DistantThunder** [reply]
Ah sorry, my bad I thought I was on ROCM repo actually, ofc ROCm distribution isn't your problem.

**Reply by vladmandic** [reply]
i'm trying to make this repo as cross-platform as possible, but for that i'm relying on contiributors a lot.
and that way we managed to get amd-rocm, apple-m1/m2, torch-directml and intel-arc and intel-ipex support, not just standard nvidia cuda.

**Reply by Soulreaver90** [reply]
@DistantThunder  I was able to successfully compile pytorch 2.0 (did not do 2.1 nightly) with Rocm 5.5. I am using a 6700 XT and did not notice a speed increase. I will try with latest 2.1 later, but this may very well be targeted towards RDNA 3. 
I basically followed what was posted this, but skipped the docker portion as ROCM 5.5 is already installed on my system.

**Reply by iDeNoh** [reply]
> @DistantThunder I was able to successfully compile pytorch 2.0 (did not do 2.1 nightly) with Rocm 5.5. I am using a 6700 XT and did not notice a speed increase. I will try with latest 2.1 later, but this may very well be targeted towards RDNA 3. I basically followed what was posted this, but skipped the docker portion as ROCM 5.5 is already installed on my system.

any chance you could link what you followed to compile? So far I've been unable to get it to work.

**Reply by Soulreaver90** [reply]
> > @DistantThunder I was able to successfully compile pytorch 2.0 (did not do 2.1 nightly) with Rocm 5.5. I am using a 6700 XT and did not notice a speed increase. I will try with latest 2.1 later, but this may very well be targeted towards RDNA 3. I basically followed what was posted this, but skipped the docker portion as ROCM 5.5 is already installed on my system.
> 
> any chance you could link what you followed to compile? So far I've been unable to get it to work.

I had meant to tag you. Umm I am not really sure how to upload it, I have to see. I want to try redoing it with the latest dev builds of pytorch, and a compile takes over an hour so ill have to find time. Who knows, maybe the official nightly will update with 5.5 faster than I do, lol.

**Reply by iDeNoh** [reply]
> > > @DistantThunder I was able to successfully compile pytorch 2.0 (did not do 2.1 nightly) with Rocm 5.5. I am using a 6700 XT and did not notice a speed increase. I will try with latest 2.1 later, but this may very well be targeted towards RDNA 3. I basically followed what was posted this, but skipped the docker portion as ROCM 5.5 is already installed on my system.
> > 
> > 
> > any chance you could link what you followed to compile? So far I've been unable to get it to work.
> 
> I had meant to tag you. Umm I am not really sure how to upload it, I have to see. I want to try redoing it with the latest dev builds of pytorch, and a compile takes over an hour so ill have to find time. Who knows, maybe the official nightly will update with 5.5 faster than I do, lol.

No worries, I *think* I was able to get it to start compiling, turns out it was picking up my old 5.4.3 library instead of the 5.5, but I'm worried its not compiling for ROCm, I'll see what it ends up doing when it finishes lol.

**Reply by Soulreaver90** [reply]
I completely removed all instances of rocm and installed 5.5 from scratch. If you look at the logs while it’s compiling, you should see 5.5 in the paths listed. You’ll also need to compile vision but it’s much faster. Also if you simply do a setup.py install, it will install in the PyTorch folder. I had to run setup.py install --prefix=/home/blah/automatic/venv

**Reply by DistantThunder** [reply]
@Soulreaver90  I wouldn't want to turn this into a ROCm support thread but I checked out v2.0.0 of pytorch, ran the requirements install and attempted to use setup.py with "develop" arg.

I have compile failures (/usr/lib/gcc/x86_64-pc-linux-gnu/12.2.1/include/avx512fintrin.h:2301:10: error: ‘__Y’ may be used uninitialized [-Werror=maybe-uninitialized]) each time I try.

Basically hitting this I believe:
* https://github.com/pytorch/pytorch/issues/77939#issuecomment-1528168416

Using gcc 13 newly released on Arch in a container, I get errors on:

> ‘uint32_t’ in namespace ‘std’ does not name a type; did you mean ‘wint_t’

I really hope we get some nightly from pytorch soon.

**Reply by YHD233** [reply]
> @DistantThunder I was able to successfully compile pytorch 2.0 (did not do 2.1 nightly) with Rocm 5.5. I am using a 6700 XT and did not notice a speed increase. I will try with latest 2.1 later, but this may very well be targeted towards RDNA 3. I basically followed what was posted this, but skipped the docker portion as ROCM 5.5 is already installed on my system.

I would like to ask how you modified the version number of the torch you built. The version I built myself is like this 2.1.0a0+git7a15e82 instead of torch2.1.0a0+rocm5.5

**Reply by evshiron** [reply]
@YHD233 

[`PYTORCH_BUILD_VERSION` and `PYTORCH_BUILD_NUMBER`](https://github.com/pytorch/pytorch/blob/74dc2a53f62980afef40f65d064d827a023725d6/tools/generate_torch_version.py#L46)
